internal/polkavm/host_call/accumulate_functions.go

```diff
@@ -253,9 +256,63 @@ func Eject(gas Gas, regs Registers, mem Memory, ctxPair AccumulateContextPair, t
 	}
 	gas -= EjectCost
 
-	// TODO: implement method
+	d, o := regs[A0], regs[A1]
 
-	return gas, regs, mem, ctxPair, nil
+	// let h = μo..o+32 if Zo..o+32 ⊂ Vμ
+	h := make([]byte, 32)
+	if err := mem.Read(uint32(o), h); err != nil {
+		// otherwise ∇
+		return gas, withCode(regs, OOB), mem, ctxPair, nil
+	}
+
+	if block.ServiceId(d) == ctxPair.RegularCtx.ServiceId {
+		// d = x_s => WHO
+		return gas, withCode(regs, WHO), mem, ctxPair, nil
+	}
+
+	// if d ∈ K((x_u)_d)
+	serviceAccount, ok := ctxPair.RegularCtx.AccumulationState.ServiceState[block.ServiceId(d)]
+	if !ok {
+		return gas, withCode(regs, WHO), mem, ctxPair, nil
+	}
+
+	encodedXs, err := jam.Marshal(struct {
```

Comment:
Why not just use the method like this? is it because of the `length=32` tag?
```suggestion
	encodedXs, err := jam.Marshal(ctxPair.RegularCtx.ServiceId)
```

Reply:
Yes the number of octets/bytes should be 32 based on GP. So I had to use the length tag